### DISCUSSION TITLE
push untracked branches?

### DISCUSSION THREAD

**Discussion by Zoybean** [jj-vcs/jj#4315] [Ideas]
Sometimes I want to push a branch to a given remote, without considering that a canonical pairing of local-remote. Could we have some `jj git push -b foo-branch --remote some-remote --allow-untracked`? Or does pushing only make sense when it applies to a local that tracks a remote?

**Comment by yuja** [comment]
There are some related issues, not implemented yet. In colocated repo, maybe you can use `git` instead.
https://github.com/martinvonz/jj/pull/2098
https://github.com/martinvonz/jj/issues/3588

**Reply by Zoybean** [reply]
in git, I think it's a bit worse. I'm not sure if I can push an untracked branch, but I do have to checkout the branch first, and my motivation for this is to avoid needing to (seemingly arbitrarily) align my local state before I can push a branch. That said, tracking a branch seems less arbitrary than checking it out.

I think #2098 might address my use-case (though from a different angle), letting me `jj git push-thing -r xyz --to-branch foo-branch --remote some-remote`
